src/porepy/applications/verification_setups/manu_poromech_nofrac.py

```diff
@@ -362,159 +466,162 @@ def flow_source(self, sd: pp.Grid, time: float) -> np.ndarray:
+        out = SaveData()
 
-        # Store time for convenience
-        self.time = t
+        # Time
+        out.time = t
 
         # Pressure
-        self.exact_pressure = ex.pressure(sd=sd, time=t)
-        self.approx_pressure = data[pp.STATE][pp.ITERATE][p_name]
-        self.error_pressure = self.relative_l2_error(
+        out.exact_pressure = self.exact_sol.pressure(sd=sd, time=t)
+        out.approx_pressure = data[pp.STATE][pp.ITERATE][p_name]
+        out.error_pressure = self.relative_l2_error(
             grid=sd,
-            true_array=self.exact_pressure,
-            approx_array=self.approx_pressure,
+            true_array=out.exact_pressure,
+            approx_array=out.approx_pressure,
             is_scalar=True,
             is_cc=True,
         )
 
-        # Darcy flux
-        self.exact_flux = ex.darcy_flux(sd=sd, time=t)
-        self.approx_flux = data[pp.STATE][pp.ITERATE][q_name]
-        self.error_flux = self.relative_l2_error(
-            grid=sd,
-            true_array=self.exact_flux,
-            approx_array=self.approx_flux,
-            is_scalar=True,
-            is_cc=False,
-        )
-
         # Displacement
-        self.exact_displacement = ex.displacement(sd=sd, time=t)
-        self.approx_displacement = data[pp.STATE][pp.ITERATE][u_name]
-        self.error_displacement = self.relative_l2_error(
+        out.exact_displacement = self.exact_sol.displacement(sd=sd, time=t)
+        out.approx_displacement = data[pp.STATE][pp.ITERATE][u_name]
+        out.error_displacement = self.relative_l2_error(
             grid=sd,
-            true_array=self.exact_displacement,
-            approx_array=self.approx_displacement,
+            true_array=out.exact_displacement,
+            approx_array=out.approx_displacement,
             is_scalar=False,
             is_cc=True,
         )
 
-        # Poroelastic force
-        self.exact_force = ex.poroelastic_force(sd=sd, time=t)
-        self.approx_force = data[pp.STATE][pp.ITERATE][force_name]
-        self.error_force = self.relative_l2_error(
+        # Flux
+        out.exact_flux = self.exact_sol.darcy_flux(sd=sd, time=t)
+        out.approx_flux = data[pp.STATE][pp.ITERATE][q_name]
+        out.error_flux = self.relative_l2_error(
             grid=sd,
-            true_array=self.exact_force,
-            approx_array=self.approx_force,
+            true_array=out.exact_flux,
+            approx_array=out.approx_flux,
+            is_scalar=True,
+            is_cc=False,
+        )
+
+        # Force
+        out.exact_force = self.exact_sol.poroelastic_force(sd=sd, time=t)
+        out.approx_force = data[pp.STATE][pp.ITERATE][force_name]
+        out.error_force = self.relative_l2_error(
+            grid=sd,
+            true_array=out.exact_force,
+            approx_array=out.approx_force,
             is_scalar=False,
             is_cc=False,
         )
 
+        return out
+
+
+class SetupUtilities:
```

Comment:
Again, move to verification_utils? Any possibility of refactoring for code reuse?

Reply:
Did not move to verification_utils, but reused code when possible.